six/projects/csm/source/SIXSensorModel.cpp

```diff
@@ -1110,5 +1110,62 @@ double SIXSensorModel::getCorrelationCoefficient(size_t cpGroupIndex,
                          "SIXSensorModel::getCorrelationCoefficient");
     }
 }
+
+DataType SIXSensorModel::getDataType(const csm::Des& des)
+{
+    // NOTE: Versions of SICD <= 1.1 and SIDD <= 1.0 prefixed FTITLE with
+    //       SICD or SIDD, so for old files we could key off of that.  Since
+    //       that's not guaranteed for newer versions though, we now use the
+    //       DESID for really old versions and the DESSHSI from
+    //       XML_DATA_CONTENT for newer versions.
+    std::string desid = des.subHeader().substr(NITF_DE_SZ, NITF_DESTAG_SZ);
+    str::trim(desid);
+    // SICD/SIDD 1.0 specify DESID as XML_DATA_CONTENT
+    // Older versions of the spec specified it as SICD_XML/SIDD_XML
+    // Here we'll accept any of these under the assumption that it's not
+    // such an old version of the spec that the XML layout itself has
+    // changed (if it did, XMLControl will end up throwing anyway)
+    if (desid == "SICD_XML")
+    {
+        return DataType::COMPLEX;
+    }
+    else if (desid == "SIDD_XML")
+    {
+        return DataType::DERIVED;
+    }
+    else if (desid == Constants::DES_USER_DEFINED_SUBHEADER_TAG)
+    {
+        // Check whether DES is SICD/SIDD, or something of a different
+        // sort. Note that we need to check the subheader
+        // length first rather than calling tre.getCurrentSize() in
+        // case there is no subheader (in which case
+        // subheader.getSubheaderFields() will throw with a
+        // NITRO-specific message).
+        const std::string& subheader = des.subHeader();
```

Comment:
XML_DATA_CONTENT is statically linked in now so this can use that now.  But can't you take it a step further and use a method from NITFReadControl so that the logic is in one spot (might need to refactor it a bit)?